### ISSUE TITLE
FormRequest should raise an error instead of submitting a first form if formname doesn't match anything

### ISSUE THREAD

**Issue by kmike** [scrapy/scrapy#1163]
I think this test is weird: https://github.com/scrapy/scrapy/blob/526aa07f44ab30e128ffb42d12143e4aac9c899e/tests/test_http_request.py#L511

This test shows a more sane behaviour: https://github.com/scrapy/scrapy/blob/526aa07f44ab30e128ffb42d12143e4aac9c899e/tests/test_http_request.py#L655

See also: https://github.com/scrapy/scrapy/pull/1137

**Comment by Curita** [comment]
Makes sense, both options should be consistent with each other, and you can define multiple forms as fallbacks using formxpath instead.

**Comment by AdrianAtZyte** [comment]
Closing after the move to [form2request](https://form2request.readthedocs.io/en/latest/usage.html) as a [replacement for `FormRequest`](https://github.com/scrapy/scrapy/pull/6438).

If you believe this issue still impacts `form2request` and should be addressed, please open an issue [upstream](https://github.com/scrapy/form2request/issues) with a link to this issue for reference.
